### PR TITLE
[W5.11][W13-A1]Raynold Ng

### PR DESCRIPTION
**Pull request by raynoldng**
Implement 2 features:

1. sort command, persons will be listed in alphabetical order
2. find command is case insensitive. John is matched by 'john' and 'John'

### CODE REVIEW

File: src/seedu/addressbook/commands/FindCommand.java [hunk 1]
```diff
@@ -50,7 +54,8 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
+            final Set<String> wordsInNameLowerCase = Utils.convertStringsToLowerCase(wordsInName);
```
Comment (dominickenn) [inline src/seedu/addressbook/commands/FindCommand.java]: Lines 56, 57 can be changed to one line. For e.g.
final Set<String> wordsInNameLowerCase = new HashSet<>(person.getName().getWordsInName().toLowerCase());

Comment (dominickenn) [inline src/seedu/addressbook/commands/FindCommand.java]: Lines 56, 57 can be changed to one line. For e.g.
final Set<String> wordsInNameLowerCase = new HashSet<>(person.getName().getWordsInName().toLowerCase());

Comment (dominickenn) [inline src/seedu/addressbook/commands/FindCommand.java]: Lines 56, 57 can be changed to one line. For e.g.
final Set wordsInNameLowerCase = new HashSet<>(person.getName().getWordsInName().toLowerCase());

File: src/seedu/addressbook/commands/FindCommand.java [hunk 2]
```diff
@@ -23,8 +24,11 @@
 
     private final Set<String> keywords;
 
+    /**
+     * all keywords are converted to lower case
+     */
     public FindCommand(Set<String> keywords) {
-        this.keywords = keywords;
+        this.keywords = Utils.convertStringsToLowerCase(keywords);
```
Comment (dominickenn) [inline src/seedu/addressbook/commands/FindCommand.java]: There is no need to call for a static method.
Use String.toLowerCase() method.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        try {
+            expected = new UniquePersonList(expectedSortedList);
+        } catch (UniquePersonList.DuplicatePersonException e) {
+            e.printStackTrace();
+        }
+
+        assertTrue(peopleInBook.equals(expected));
+    }
+
+}
```
Comment (dominickenn) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:41]: Could apply more SLAP to SortCommandTest.
Refer to addCommandTest to see abstraction of codes into methods.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 4]
```diff
@@ -3,8 +3,7 @@
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
-import java.util.Arrays;
-import java.util.List;
+import java.util.*;
```
Comment (dominickenn) [inline test/java/seedu/addressbook/common/UtilsTest.java:6]: Not right to do java.util.*. Should be added as such : java.util.HashSet;

File: src/seedu/addressbook/common/Utils.java [hunk 5]
```diff
@@ -34,4 +34,15 @@ public static boolean elementsAreUnique(Collection<?> items) {
         }
         return true;
     }
+
+    /**
+     * Returns new set with all strings converted to lower case
+     */
+    public static Set<String> convertStringsToLowerCase(Set<String> keywords) {
```
Comment (okkhoy) [inline src/seedu/addressbook/common/Utils.java:41]: Good use of static method!

File: test/expected.txt [hunk 6]
```diff
@@ -297,6 +300,42 @@
...
+|| 	5. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| 
+|| 5 persons listed!
+|| ===================================================
+|| Enter command: || [Command entered:  sort]
+|| All persons in address book sorted.
+|| ===================================================
+|| Enter command: || [Command entered:  list]
+|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| 	2. Betsy Choo Tags: [secretive]
```
Comment (okkhoy) [inline test/expected.txt:332]: Good test case to show the before and after `sort`

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+package seedu.addressbook.commands;
+
+import org.junit.Test;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.util.TestUtil;
+
+import java.util.*;
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:9]: Don't import everything using *; import only the necessary classes explicitly.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+import static org.junit.Assert.*;
+
+public class SortCommandTest {
+
+    @Test
+    public void execute_sortCommand_threePersons() {
+        SortCommand command = new SortCommand();
+
+        Person alice = TestUtil.generateTestPerson("Alice");
```
Comment (okkhoy) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:19]: Going further you can consider moving all the steps necessary for the test to run (or preconditions) to a setup method before beginning the tests; correspondingly can put all the tear down in a method after the tests.

**Comment by okkhoy** [comment]
@raynoldng 
Some comments added. Please ack & close the PR after reading comments.

**Comment by raynoldng** [comment]
Acknowledged with thanks

**Pull request closed without merging**